GCS/probe0/probe0.go

```diff
@@ -0,0 +1,142 @@
+import (
+	"context"
+	"fmt"
+	"sync"
+	"io"
+	"os"
+	"time"
+	"cloud.google.com/go/storage"
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/metrics"
+	"github.com/google/cloudprober/probes/options"
+	"github.com/google/cloudprober/targets/endpoint"
+	"github.com/googleinterns/step224-2020/blob/evan/hermesMain/cmd"
+)
+
+// Probe holds aggregate information about all probe runs, per-target.
+type Probe struct {
+	name    string
+	c       *ProbeConf
+	targets []string
+	opts    *options.Options
+
+	res map[string]*metrics.EventMetrics // Results by target
+	l   *logger.Logger
+}
+
+// Init initializes the probe with the given params.
+func (p *Probe) Init(name string, opts *options.Options) error {
+	c, ok := opts.ProbeConf.(*ProbeConf)
```

Comment:
This will run as the proto message for `Probe0` is `ProbeConf`.

This line is casting an interface public field of a struct when accessing it (I think).

The field is defined here: https://github.com/google/cloudprober/blob/dd0bb3e2e8a54038da64c9b092db2f8941ecb69c/probes/options/options.go#L40